### PR TITLE
Added two methods incrementEach and decrementEach

### PR DESCRIPTION
**Pull request by SmallRuralDog**
add incrementEach
add decrementEach

### CODE REVIEW

**Comment by GromNaN** [comment]
Hello, thanks for the suggestion. For reference, this methods were added in Laravel 9 by https://github.com/laravel/framework/pull/45577.

Would you like to rebase the PR, add tests in [`QueryBuilderTest`](https://github.com/jenssegers/laravel-mongodb/blob/master/tests/QueryBuilderTest.php) and update the changelog?

File: src/Query/Builder.php [hunk 1]
```diff
@@ -647,6 +647,29 @@ public function increment($column, $amount = 1, array $extra = [], array $option
...
+        if (! empty($extra)) {
+            $query['$set'] = $extra;
+        }
+
+        // Protect
+        foreach ($columns as $column => $amount) {
+            $this->where(function ($query) use ($column) {
+                $query->where($column, 'exists', false);
+
+                $query->orWhereNotNull($column);
```
Comment (GromNaN) [inline src/Query/Builder.php:664]: With cumulative conditions on all fields, no field will be incremented if at least 1 of the fields is `null`.

We'd be wise not to set any conditions at all and let the server return an error.

**Review by alcaeus — CHANGES_REQUESTED** [review]
Please address the comment by @GromNaN and add tests for the new functionality.

**Review by GromNaN — APPROVED** [review]

**Comment by GromNaN** [comment]
PR updated with tests.

File: src/Query/Builder.php [hunk 2]
```diff
@@ -789,24 +789,18 @@ public function increment($column, $amount = 1, array $extra = [], array $option
...
-        // Protect
+        // Not using $inc for each column, because it would fail if one column is null.
         foreach ($columns as $column => $amount) {
-            $this->where(function ($query) use ($column) {
-                $query->where($column, 'exists', false);
-
-                $query->orWhereNotNull($column);
-            });
+            $stage['$addFields'][$column] = [
+                '$add' => [['$ifNull' => ['$' . $column, 0]], $amount],
```
Comment (alcaeus) [inline src/Query/Builder.php]: According to the [docs](https://www.mongodb.com/docs/manual/reference/operator/aggregation/add/), there's an optimisation we can leverage in MongoDB 6.1 and later:
> Starting in MongoDB 6.1 you can optimize the `$add` operation. To improve performance, group references at the end of the argument list.

```suggestion
                '$add' => [$amount, ['$ifNull' => ['$' . $column, 0]]],
```

**Review by alcaeus — COMMENTED** [review]
Minor suggestion to optimise the `$add` operator, but I like that the behaviour now better matches the expectation from Laravel that all documents are updated and we gracefully handle `null` values.

File: src/Query/Builder.php (hunk 2 above)
Comment (GromNaN) [reply src/Query/Builder.php]: Thanks. It's strange that the server isn't able to apply this kind of optimisation itself.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**